### ISSUE TITLE
Not getting messages on Telegram

### ISSUE THREAD

**Issue by x2611** [tommyblue/ED-AFK-Notifier#11]
Hi,

Set up bot as instructed. Made Telegram bot, added tokens, paths, IDs to config. Seems to be alright.
I get the following message after /check
If you received this message, everything is configured properly! :)

Sitting in ship at REZ, start bot, it says:
2022/06/01 13:44:36 Monitoring the journal file at C:\Users\my_name\Saved Games\Frontier Developments\Elite Dangerous\Journal.220601112123.01.log

Telegram running with bot channel open but I get no messages. 15 ships killed so far.
Tried both versions.
ED-AFK-Notifier_0.3.3_Windows_i386
ED-AFK-Notifier_0.3.3_Windows_x86_64
Win10 + ED-Horizons. (Tried with Odyssey and RunAsAdmin too. No go)
Turned on debug and it shows it's reading the log in the console.

What am I doing wrong?
Thanks

**Comment by tommyblue** [comment]
@x2611 could you paste here your configuration file (removing secret tokens)?

**Comment by x2611** [comment]
[journal]
    Path = "C:\\Users\\my_name\\Saved Games\\Frontier Developments\\Elite Dangerous"
    debug = false # Print a logline for each new line in the journal file
    fighter = true # When true, send notification also for hull damage to the fighter
    shields = true # When true, send a notification when shields state changes (up/down)
    kills = true # When true, send notification on each new kill, including total reward earned (noisy!)
    silent_kills = false # When true, reduce noise for kill notification, sending a notification every 10 kills

[telegram]
    token = "1234567ID"
    channelId = 1234567

**Comment by tommyblue** [comment]
strange, I need to check. Thanks

**Comment by apattillo** [comment]
For what it's worth (I know nothing about go), I was having the same problem with the windows executable downloaded from releases. However, when I built the executable myself, it worked great. 

```
PS C:\Users\alpat\source\repos\ED-AFK-Notifier\cmd\ed-afk-notifier> go build
PS C:\Users\alpat\source\repos\ED-AFK-Notifier\cmd\ed-afk-notifier> .\ed-afk-notifier.exe
time="2022-06-09T22:45:06-05:00" level=info msg="Config:"
time="2022-06-09T22:45:06-05:00" level=info msg="  Notify fighter status: true"
time="2022-06-09T22:45:06-05:00" level=info msg="  Notify shields status: true"
time="2022-06-09T22:45:06-05:00" level=info msg="  Notify on kills: true (silent: false)"
time="2022-06-09T22:45:06-05:00" level=info msg="  Journal file path: C:\\Users\\alpat\\Saved Games\\Frontier Developments\\Elite Dangerous"
time="2022-06-09T22:45:07-05:00" level=info msg="Found most recent journal file: Journal.220609150554.01.log"
time="2022-06-09T22:45:07-05:00" level=info msg="Reading journal..."
```
![image](https://user-images.githubusercontent.com/133790/172986465-e7999f28-5b4a-4058-afcf-205582554d43.png)

**Comment by x2611** [comment]
> For what it's worth (I know nothing about go), I was having the same problem with the windows executable downloaded from releases. However, when I built the executable myself, it worked great.

Interesting. I believe you. I had the same thing happen with a Linux program years ago. Depo version would always crash to desktop immediately but the compiled source worked great. Computers are weird. 

I might try this myself. Downloading Go now. Feeling hacky today lol.

**Comment by x2611** [comment]
OK I compiled it and it's running. Need to grab some missions for testing. Comparing the compiled EXE to the two from releases, the sizes are all way different. What's up with that? And how do I know which one I built? The 32 or 64 bit? Thx.

![Untitled](https://user-images.githubusercontent.com/60093096/172998105-7fe7b01f-c19a-43e7-bade-4527e2069b4c.png)

**Comment by x2611** [comment]
Well nope. Compiled version don't work either :(

![Untitled2](https://user-images.githubusercontent.com/60093096/173006967-c27d9396-8939-489b-b823-d923013ae48e.png)
![Screenshot_20220610-083947_Telegram](https://user-images.githubusercontent.com/60093096/173007214-7ba12898-f46e-4adb-bc5e-3d625845c920.jpg)

**Comment by tommyblue** [comment]
definitely I confirm the issue. I think it's caused by the most recent changes, I changed how the log files are parsed to be able to get the correct kills count even if the program is launched after the game has started. I'll take a look during the next days

**Comment by x2611** [comment]
> definitely I confirm the issue. I think it's caused by the most recent changes, I changed how the log files are parsed to be able to get the correct kills count even if the program is launched after the game has started.

Ok, good news. I hope you can fix it. But I thought we were supposed to start game first anyway so ED-AFK grabs the actual log file.

**Comment by tommyblue** [comment]
Yes, though I was working on auto loading new and more recent log files

Il ven 10 giu 2022, 17:46 x2611 ***@***.***> ha scritto:

**Comment by x2611** [comment]
Oh Ok, I understand. That sounds like a very nice feature. Anyways, I will be happy to help you test it going along. Let me know. X.

**Comment by tommyblue** [comment]
Hi folks, sorry I didn't play ED in the last months. Today I played a bit and I tried to update some libs. I wasn't able to reproduce the issue, I always received the telegram messages (trying on linux). But the telegram lib had recent changes so I tried to update it.
I drafted a new release, it also contains other improvements I was trying to add. They're probably still a bit broken but some of them are cool (like auto-loading the new journal files when it changes, like when you exit the game then restart it). Let's give it a try and let me know if it fixes the issue 🙏🏼  
https://github.com/tommyblue/ED-AFK-Notifier/releases/tag/v0.4.0-alpha

**Comment by x2611** [comment]
Thanks Tommy, will try it out later.

**Comment by tommyblue** [comment]
ok thanks. I gave it a try on windows using the released binary and I received the messages correctly
